### DISCUSSION TITLE
请问默认的数据文件目录是什么？谢谢

### DISCUSSION THREAD

**Discussion by shyshi** [usememos/memos#64] [Q&A] [answered]
您好，采用 docker hub 里面的映像（https://registry.hub.docker.com/r/neosmemo/memos/）
部署到群晖上，并不通过命令，而是通过群晖自带的设置启动，所以无法设置数据目录（/path/to/data:/var/opt/memo）。群晖支持定义数据目录功能，但是即使将 /var/opt/memos映射到NAS中的硬盘目录，一旦重启docker，则数据和用户注册信息全部丢失，看起来可能是因为默认数据路径有误，所以请问如果指定数据目录，默认的数据位置在哪里？谢谢。（版本 0.0.1）

**Comment by boojack** [comment, marked as answer]
```bash
docker run --name memos --publish 5230:8080 --volume ~/.memos/:/var/opt/memos -e mode=prod neosmemo/memos:0.0.1
```
当 `mode` 指定为 `prod` 时，数据库文件位置应该是在 `/var/opt/memos`，否则在 `/usr/local/memos`。

**Reply by boojack** [reply]
而且看上去是可以设置环境变量的：https://kb.synology.com/en-af/DSM/tutorial/How_can_I_debug_and_fix_common_Docker_container_issues
<img width="822" alt="image" src="https://user-images.githubusercontent.com/24653555/169642860-064a6870-e19c-4089-9b5b-adfec53d7167.png">

**Reply by shyshi** [reply]
谢谢。搞定了。群晖上在非prod模式下映射目录有问题，不允许映射，必须在“Environment Variables“ 里面加上 mode:prod 的设置，然后把本地目录映射到 /var/opt/memos 就可以解决 docker 重启数据丢失的问题了。
